### PR TITLE
chore: bump deps and refactor eip-3074 test

### PR DESCRIPTION
**Pull request by fgimenez**
Bumps reth, alloy and revm-inspectors deps.

With the updated reth, after https://github.com/paradigmxyz/reth/pull/7787 `DEV` chainspec has the OP hardforks enabled and the previous `L1BlockInfoError` is also fixed, so here the eip3074 integration test is refactored to use it.

### CODE REVIEW

File: crates/testing/src/tests.rs [hunk 1]
```diff
@@ -88,9 +81,15 @@ async fn test_eip3074_integration() {
     let signer_wallet = Wallet::random();
     let signer_account: EthereumSigner = signer_wallet.clone().into();
     let signer_address = signer_account.default_signer().address();
-    let signer_balance = provider.get_balance(signer_address, None).await.unwrap();
+    let signer_balance = provider
+        .get_balance(signer_address, BlockId::Number(BlockNumberOrTag::Latest))
+        .await
```
Comment (mattsse) [inline crates/testing/src/tests.rs:86]: there's BlockId::latest()

**Review by mattsse — APPROVED** [review]
lgtm

hint for BlockId::Latest

File: crates/testing/src/tests.rs (hunk 1 above)
Comment (fgimenez) [reply crates/testing/src/tests.rs]: nice, thanks!

**Review by Rjected — APPROVED** [review]
nice, lgtm

**Pull request merged**